File: typescript-playwright/src/clients/AdminCommentClient.ts
import type { AdminCommentApi } from '../api/AdminCommentApi.ts';
import type { 
  AdminCommentDto, 
  AdminCommentsPageResponse, 
  CommentStatsDto, 
  CommentStatus 
} from '../utils/types.ts';
import { statusIn } from '../utils/statusIn.ts';
import { HTTP } from '../utils/constants.ts';
import { step } from '../utils/decorators.ts';

export class AdminCommentClient {
  constructor(private readonly api: AdminCommentApi) {}

  @step()
  async listAll(page = 0, size = 20): Promise<AdminCommentsPageResponse> {
    const res = await this.api.getComments(page, size);
    if (!statusIn(HTTP.OK)(res.status())) {
      throw new Error(`[AdminCommentClient] listAll failed: ${res.status()} ${await res.text()}`);
    }
    return res.json() as Promise<AdminCommentsPageResponse>;
  }

  @step()
    async getStats(): Promise<CommentStatsDto> {
      const res = await this.api.getStats();
      if (!statusIn(HTTP.OK)(res.status())) {
        throw new Error(`[AdminCommentClient] getStats failed: ${res.status()} ${await res.text()}`);
      }
      return res.json() as Promise<CommentStatsDto>;
    }

  @step()
  async moderate(
    commentId: number,
    status: CommentStatus,
    rejectionReason = ''
  ): Promise<AdminCommentDto> {
    const res = await this.api.moderate(commentId, status, rejectionReason);
    if (!statusIn(HTTP.OK, HTTP.NO_CONTENT)(res.status())) {
      throw new Error(
        `[AdminCommentClient] moderate(${commentId}) failed: ${res.status()} ${await res.text()}`
      );
    }
    return res.json() as Promise<AdminCommentDto>;
  }

  @step()
  async delete(commentId: number): Promise<void> {
    const res = await this.api.deleteComment(commentId);
    if (!statusIn(HTTP.NO_CONTENT, HTTP.OK)(res.status())) {
      throw new Error(
        `[AdminCommentClient] delete(${commentId}) failed: ${res.status()} ${await res.text()}`
      );
    }
  }
}

File: python-pytest/src/clients/admin_comment_client.py
from __future__ import annotations

from typing import cast

from src.api.admin_comment_api import AdminCommentApi
from src.utils.constants import HTTP
from src.utils.decorators import step
from src.utils.types import (
    AdminCommentDto,
    AdminCommentsPageResponse,
    CommentStatsDto,
    CommentStatus,
)


class AdminCommentClient:
    def __init__(self, api: AdminCommentApi) -> None:
        self._api = api

    @step()
    def list_all(self, page: int = 0, size: int = 20) -> AdminCommentsPageResponse:
        response = self._api.get_comments(page, size)
        if response.status_code not in (HTTP.OK,):
            raise RuntimeError(
                f"[AdminCommentClient] listAll failed: {response.status_code} {response.text}"
            )
        return cast(AdminCommentsPageResponse, response.json())

    @step()
    def get_stats(self) -> CommentStatsDto:
        response = self._api.get_stats()
        if response.status_code not in (HTTP.OK,):
            raise RuntimeError(
                f"[AdminCommentClient] getStats failed: {response.status_code} {response.text}"
            )
        return cast(CommentStatsDto, response.json())

    @step()
    def moderate(
        self, comment_id: int, status: CommentStatus, rejection_reason: str = ""
    ) -> AdminCommentDto:
        response = self._api.moderate(comment_id, status, rejection_reason)
        if response.status_code not in (HTTP.OK, HTTP.NO_CONTENT):
            raise RuntimeError(
                f"[AdminCommentClient] moderate({comment_id}) failed: {response.status_code} {response.text}"
            )
        return cast(AdminCommentDto, response.json())

    @step()
    def delete(self, comment_id: int) -> None:
        response = self._api.delete_comment(comment_id)
        if response.status_code not in (HTTP.NO_CONTENT, HTTP.OK):
            raise RuntimeError(
                f"[AdminCommentClient] delete({comment_id}) failed: {response.status_code} {response.text}"
            )
